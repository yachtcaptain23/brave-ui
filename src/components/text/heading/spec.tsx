/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Heading from './index'

describe('all heading tests', () => {
  const baseHeadingComponent = (props?: object) => (
    <Heading id='testHeading' {...props} />
  )
  describe('heading tests', () => {
    describe('basic tests', () => {
      it('matches the snapshot', () => {
        const component = baseHeadingComponent()
        const tree = create(component).toJSON()
        expect(tree).toMatchSnapshot()
      })

      it('renders the component', () => {
        const wrapper = shallow(baseHeadingComponent({id: 'toughTitleHeading'}))
        const assertion = wrapper.find('#toughTitleHeading').length
        expect(assertion).toBe(1)
      })
    })

    describe('component behavior', () => {
      it('can have an id', () => {
        const wrapper = shallow(baseHeadingComponent({id: 'titleHeadingThing'}))
        const assertion = wrapper.props().id
        expect(assertion).toBe('titleHeadingThing')
      })

      it('can be an h1 when level === 1', () => {
        const wrapper = shallow(baseHeadingComponent({level: 1}))
        const assertion = wrapper.html().includes('h1')
        expect(assertion).toBe(true)
      })

      it('can be an h2 when level === 2', () => {
        const wrapper = shallow(baseHeadingComponent({level: 2}))
        const assertion = wrapper.html().includes('h2')
        expect(assertion).toBe(true)
      })

      it('can be an h3 when level === 3', () => {
        const wrapper = shallow(baseHeadingComponent({level: 3}))
        const assertion = wrapper.html().includes('h3')
        expect(assertion).toBe(true)
      })

      it('can pass text', () => {
        const wrapper = shallow(baseHeadingComponent({text: 'nsync rlz'}))
        const assertion = wrapper.html().includes('nsync rlz')
        expect(assertion).toBe(true)
      })
    })

    describe('theming', () => {
      it('allows theming the `color` property', () => {
        const component = baseHeadingComponent({theme: { color: 'orange' } })
        const tree = create(component).toJSON()
        expect(tree).toHaveStyleRule('color', 'orange')
      })

      it('allows theming the `margin` property', () => {
        const component = baseHeadingComponent({theme: { margin: '30px' } })
        const tree = create(component).toJSON()
        expect(tree).toHaveStyleRule('margin', '30px')
      })
    })
  })
})
