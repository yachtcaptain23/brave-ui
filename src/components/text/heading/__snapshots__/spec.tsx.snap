// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all heading tests heading tests basic tests matches the snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  font-family: "Poppins",sans-serif;
  -webkit-font-smoothing: antialiased;
  color: #4b4c5c;
  margin: 0;
  font-size: 28px;
  font-weight: 600;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

<h1
  className="c0"
  id="testHeading"
/>
`;
